### ISSUE TITLE
CI: GKE test fail due to metrics-server in ContainerCreating state, Cilium in CrashLoopBackOff

### ISSUE THREAD

**Issue by jrajahalme** [cilium/cilium#11286]
Cilium agents are in CrashLoopBackOff with this error in logs:
```
2020-05-01T04:34:49.187829329Z level=fatal msg="Unable to initialize Kubernetes subsystem" error="unable to retrieve the complete list of server APIs: metrics.k8s.io/v1beta1: the server is currently unable to handle the request" subsys=daemon
```

While metrics-server and many other kube-system PODs have never started:
```
	 cilium        cilium-c8l4l                                                0/1     CrashLoopBackOff    6          11m   10.138.15.192   gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 cilium        cilium-node-init-mk6wn                                      1/1     Running             0          30h   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 cilium        cilium-node-init-ml7lc                                      1/1     Running             0          30h   10.138.15.192   gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 cilium        cilium-operator-74c5c88969-6l49x                            0/1     CrashLoopBackOff    6          11m   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 cilium        cilium-q67rd                                                0/1     CrashLoopBackOff    6          11m   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 cilium        log-gatherer-2nsqm                                          1/1     Running             13         36h   10.138.15.192   gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 cilium        log-gatherer-5ttmk                                          1/1     Running             12         36h   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 default       test-k8s2-664d69b864-s7n56                                  1/2     Running             0          31h   10.41.35.139    gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 default       testclient-6ndjf                                            1/1     Running             0          31h   10.41.34.97     gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 default       testclient-t59tz                                            1/1     Running             0          31h   10.41.35.143    gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 default       testds-2p98k                                                1/2     Running             0          31h   10.41.34.73     gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 default       testds-5x8n5                                                1/2     Running             0          31h   10.41.35.129    gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   event-exporter-v0.2.5-7df89f4b8f-scf2l                      0/2     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   fluentd-gcp-scaler-54ccb89d5-qvz8x                          0/1     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   fluentd-gcp-v3.1.1-5rf7h                                    2/2     Running             0          10m   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   fluentd-gcp-v3.1.1-wjgw9                                    2/2     Running             0          10m   10.138.15.192   gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   heapster-gke-5858fc79b6-vqhb2                               0/3     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   kube-dns-5877696fb4-7rcq2                                   0/4     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   kube-dns-5877696fb4-8vlxk                                   0/4     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   kube-dns-autoscaler-8687c64fc-npsg8                         0/1     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   kube-proxy-gke-cilium-ci-2-default-pool-43b1a44e-lz2w       1/1     Running             0          10m   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   kube-proxy-gke-cilium-ci-2-default-pool-43b1a44e-q3gw       1/1     Running             0          10m   10.138.15.192   gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   l7-default-backend-8f479dd9-2zq8h                           0/1     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   metrics-server-v0.3.1-5c6fbf777-w7pwd                       0/2     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   prometheus-to-sd-jppmc                                      2/2     Running             0          10m   10.138.15.192   gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   prometheus-to-sd-v5t8q                                      2/2     Running             0          10m   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   registry-adder-fmbrl                                        1/1     Running             0          10m   10.138.15.192   gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
	 kube-system   registry-adder-sn2gb                                        1/1     Running             0          10m   10.138.0.9      gke-cilium-ci-2-default-pool-43b1a44e-lz2w   <none>           <none>
	 kube-system   stackdriver-metadata-agent-cluster-level-6588cdf48b-4mndx   0/2     ContainerCreating   0          10m   <none>          gke-cilium-ci-2-default-pool-43b1a44e-q3gw   <none>           <none>
```

Seems like we should wait for the k8s deployment to be up-and-running before starting Cilium, or at least that CIlium is up-and-running before starting the tests?

**Comment by joestringer** [comment]
I also hit this in local testing.

**Comment by joestringer** [comment]
Duplicate of #11262 as far as I can tell. Feel free to reopen if you disagree.
